### ISSUE TITLE
Question regarding manual commit example

### ISSUE THREAD

**Issue by Iadgarov** [twmb/franz-go#801]
Hello, 

I've been going over the manual commit example [provided in the repo](https://github.com/twmb/franz-go/blob/master/examples/goroutine_per_partition_consuming/manual_commit/main.go). 

What I don't fully understand is the purpose of creating a buffered channel of 5 for each partitions consumer `recs` input, where each item in the channel is a batch of messages.
I am referring to [this line](https://github.com/twmb/franz-go/blob/6b61d17383b1e0c766143118efa39f1f320a3b2b/examples/goroutine_per_partition_consuming/manual_commit/main.go#L71).

It seems like if we give each of the partition consumers a buffered channel then in theory the main poll loop could send the same batch of messages to the consumer several times?

Lets say we go into the first iteration of the [poll loop](https://github.com/twmb/franz-go/blob/6b61d17383b1e0c766143118efa39f1f320a3b2b/examples/goroutine_per_partition_consuming/manual_commit/main.go#L138). 
We 
* poll for records
* send the relevant ones to each of the partition consumers, and since the channels are buffered we do not block or wait for them to complete their processing. 
* With processing not yet done, the manual commit is not yet triggered, however we can already poll Kafka for more messages, once again sending them down the channel to each of the partition consumers. 

Would this second batch not be identical to the first due to the potential lack of commit between the two polls? 
Also, in such a case would we not be allowing a rebalance to occur potentially before processing is done?

tl;dr - shouldn't we want [fetches.EachPartition](https://github.com/twmb/franz-go/blob/6b61d17383b1e0c766143118efa39f1f320a3b2b/examples/goroutine_per_partition_consuming/manual_commit/main.go#L155) to be blocking to avoid rebalance and polling until we commit what we've already polled?
And if so, what's the benefit of using goroutine-per-partition consumers rather than a bunch of go routines handling a batch from all partitions, allowing a rebalance+new poll only after they finish (since if we need to wait for them all then maintaining partition order shouldn't really matter)?

I feel like I'm missing something fundamental, would greatly appreciate your input.
